lizard_connector/parsers.py

```diff
@@ -27,4 +51,156 @@ def parse_uuid(json_, endpoint=None):
         A list of all uuid elements in the root of the results attribute.
     """
     uuid = 'unique_id' if endpoint == 'organisations' else 'uuid'
-    return parse_element(json_, uuid)
+    return list_on_key(results, uuid)
+
+
+def flatten_dict(results, parent_key='', sep='__'):
+    """
+    Flatten dictionary.
+
+        {'a': 1,
+         'c': {'a': 2,
+               'b': {'x': 5,
+                     'y' : 10}},
+         'd': [1, 2, 3]}
+
+    Is flattened to:
+
+        {'a': 1,
+         'c__a': 2,
+         'c__b__x': 5,
+         'c__b__y': 10,
+         'd': [1, 2, 3]}
+
+    Based on:
+        https://stackoverflow.com/questions/6027558/
+        flatten_dict-nested-python-dictionaries-compressing-keys#answer-6027615
+
+    Args:
+        results(dict): multilevel dictionary.
+        parent_key(str): key to use
+        sep(str): seperator between
+    Returns:
+        Flattened dictionary.
+    """
+    items = []
+    for k, v in results.items():
+        new_key = parent_key + sep + k if parent_key else k
+        if isinstance(v, collections.MutableMapping):
+            items.extend(flatten_dict(v, new_key, sep=sep).items())
+        else:
+            items.append((new_key, v))
+    return dict(items)
+
+
+def flatten_result(results, parent_key='', sep='__'):
+    # First remove the results.
+    events = {}
+    for data_type in DATA_TYPE_FIELDS:
+        try:
+            events = results.pop(data_type) or {}
```

Comment:
Hier wordt results gemutate, hoort dat? Beetje onverwacht.

Reply:
We serveren data en metadata in de lizard-nxt api. Het idee van die scientific parser is dat die de data en metadata gescheiden ophaalt. Die `DATA_TYPE_FIELDS` bevatten data. en worden er dus vanaf gehaald.

Reply:
Dat snap ik, maar het punt was meer dat een dictionary die meegegeven wordt als parameter gemutate wordt. Dat kan verrassend zijn op de plek waar de functie aangeroepen wordt als die dictionary ook nog voor iets anders gebruikt wordt. In het algemeen kan er dan denk ik beter eerst een kopie van de dict gemaakt worden. Maar goed, komt wel.